src/main/kotlin/io/snyk/plugin/SnykBulkFileListener.kt

```diff
@@ -45,7 +46,13 @@ class SnykBulkFileListener : BulkFileListener {
 
         if (isContainerEnabled()) {
             for (project in ProjectUtil.getOpenProjects()) {
-                getKubernetesImageCache(project).extractFromEvents(events)
+                try {
+                    getKubernetesImageCache(project).extractFromEvents(events)
+                } catch (e: ClassCastException) {
+                    // if this exception occurs, IntelliJ is in an undefined state.
+                    // Outside of tests, this should not occur
+                    Logger.getInstance(javaClass).warn(e)
```

Comment:
looks like ugly workariund... could we found the resaon for that ClassCastException in tetst instead?

Reply:
And doesn't work. I can't reproduce it locally - it doesn't happen even if I test the action via act -j test

Reply:
Feel free to have a look - it has nothing to do with the actual PR functionality.

Reply:
I guess, we have to add a flag to enable/disable the filelistener with default to enable, to be able to disable it for certain tests.

Reply:
Also, the reason is quite clear - we try to access ProjectRootManager when it is not available during tests. Maybe we need to make it dumb-aware or such.